src/Http/Handler/ResponseHandlerError.php

```diff
@@ -60,15 +61,15 @@ public function execute(ResponseInterface $response)
                     $errorDescriptions[] = $error[self::FIELD_ERROR_DESCRIPTION];
                 }
 
-                throw new ApiException(
+                throw ExceptionFactory::createExceptionForResponse(
                     self::ERROR_FROM_JSON,
```

Comment:
I know it is not yours, but please make the interface of `createExceptionForResponse` consistent with that in all the other SDKs if possible :)